consensus/beacon/consensus.go

```diff
@@ -382,8 +381,6 @@ func (beacon *Beacon) FinalizeAndAssemble(chain consensus.ChainHeaderReader, hea
 	// Create the block witness and attach to block.
 	// This step needs to happen as late as possible to catch all access events.
 	if chain.Config().IsVerkle(header.Number, header.Time) {
-		// TODO(gballet) move this to the end of the overlay conversion function in a subsequent PR
-		statedb.Database().(*state.CachingDB).SaveTransitionState(header.Root, &overlay.TransitionState{Ended: true})
```

Comment:
`&overlay.TransitionState{Ended: true})` is apparently wrong here. Let's remove it for now.